### PR TITLE
Refs #17283 - fix rescuing from InThreadExecutor

### PR DESCRIPTION
**Pull request by iNecas**
+ improvements in terminate interface of in thread world

### CODE REVIEW

File: lib/dynflow/executors/abstract.rb [hunk 1]
```diff
@@ -9,9 +9,13 @@ def initialize(world)
         @logger = world.logger
       end
 
+      # @param execution_plan_id [Number] id of execution plan
```
Comment (adamruzicka) [inline lib/dynflow/executors/abstract.rb]: Isn't `execution_plan_id` a string?

File: test/testing_test.rb [hunk 2]
```diff
@@ -213,11 +213,32 @@ def plan_input
         end
       end
 
-      it "is able to handle when events ===" do
+      it "is able to handle when events" do
```
Comment (adamruzicka) [inline test/testing_test.rb]: How is this test different from [this one](https://github.com/iNecas/dynflow/blob/21173d948e3fc40fc5d05015595240fc738de656/test/testing_test.rb#L204)?

File: test/testing_test.rb [hunk 3]
```diff
@@ -213,11 +213,32 @@ def plan_input
...
+
+      describe 'auto rescue' do
+        let(:world) do
+          WorldFactory.create_world(Dynflow::Testing::InThreadWorld) do |config|
+            config.auto_rescue = true
+          end
+        end
+
+        describe 'of plan with skips' do
+
```
Comment (adamruzicka) [inline test/testing_test.rb]: This empty line probably shouldn't be here

File: lib/dynflow/executors/abstract.rb (hunk 1 above)
Comment (iNecas) [inline lib/dynflow/executors/abstract.rb]: It is! :)

File: test/testing_test.rb (hunk 2 above)
Comment (iNecas) [inline test/testing_test.rb]: Apparently, Ruby is not as smart as it looks like, and can't tell what  I want to do just from the string description. I forgot to update the plan I wanted to actually test, and it turned out I had some issues with the event handling in the in thread executor, that should now be solved

File: test/testing_test.rb (hunk 3 above)
Comment (iNecas) [inline test/testing_test.rb]: done

**Comment by iNecas** [comment]
Thanks @adamruzicka for very valuable review! The things should be fixed now

File: lib/dynflow/testing/in_thread_executor.rb [hunk 4]
```diff
@@ -7,16 +7,16 @@ def initialize(world)
         @work_items = Queue.new
       end
 
-      def execute(execution_plan_id, finished)
+      def execute(execution_plan_id, finished, _wait_for_acceptance = true)
```
Comment (adamruzicka) [inline lib/dynflow/testing/in_thread_executor.rb]: Shouldn't there be a default value for `finished` to have the interface exactly the same as the `abstract` and `parallel` executors have?

Comment (iNecas) [inline lib/dynflow/testing/in_thread_executor.rb]: Updated

**Comment by adamruzicka** [comment]
Alright, let's see what Mr. Travis has to say and get this in

**Comment by ares** [comment]
Mr. Travis is unhappy on just one worker, hopefully unrelated.

**Comment by adamruzicka** [comment]
We're green now

**Pull request merged**